discovery/scaleway/scaleway.go

```diff
@@ -0,0 +1,213 @@
+// Copyright 2021 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package scaleway
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/prometheus/common/config"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/common/version"
+	"github.com/prometheus/prometheus/discovery"
+	"github.com/prometheus/prometheus/discovery/refresh"
+	"github.com/prometheus/prometheus/discovery/targetgroup"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+// metaLabelPrefix is the meta prefix used for all meta labels.
+// in this discovery.
+const (
+	metaLabelPrefix = model.MetaLabelPrefix + "scw_"
+	separator       = ","
+)
+
+// DefaultSDConfig is the default Scaleway Service Discovery configuration.
+var DefaultSDConfig = SDConfig{
+	Zone:            "fr-par-1",
+	Port:            80,
+	RefreshInterval: model.Duration(60 * time.Second),
+}
+
+type SDConfig struct {
+	// Project: The Scaleway Project ID
+	Project string `yaml:"project"`
+
+	// Zone: The zone of the scrape targets.
+	// If you need to configure multiple zones use multiple scaleway_sd_configs
+	Zone string `yaml:"zone"`
+	// Access Key
+	AccessKey string `yaml:"access_key"`
+	// Secret Key
+	SecretKey string `yaml:"secret_key"`
```

Comment:
This should be a config.Secret field.

Reply:
Could you look at the marathon SD, there we have a custom roundtripper to also read a key from a file. I think to be consistent with other SD's we should do this (having both fields).

Reply:
Migrated to config.Secret :) for having the key from a file let's do it later as you mentioned, everything should be self-contained in the Prometheus file.